engine/gamesys/src/gamesys/resources/res_textureset.cpp

```diff
@@ -41,66 +42,75 @@ namespace dmGameSystem
-                    norm_points[i*2+1] = (points[i*2+1]) * recip_tile_height - 0.5f;
-                }
-                tile_set->m_HullSet = dmPhysics::NewHullSet2D(context, norm_points, n_points, hull_descs, n_hulls);
-                delete [] hull_descs;
-                delete [] norm_points;
+                dmGameSystemDDF::ConvexHull* hull_ddf = &texture_set_ddf->m_ConvexHulls[i];
+                tile_set->m_HullCollisionGroups[i] = dmHashString64(hull_ddf->m_CollisionGroup);
+                hull_descs[i].m_Index = (uint16_t)hull_ddf->m_Index;
+                hull_descs[i].m_Count = (uint16_t)hull_ddf->m_Count;
             }
-
-            uint32_t n_animations = texture_set_ddf->m_Animations.m_Count;
-            tile_set->m_AnimationIds.Clear();
-            // NOTE: 37 is rather arbitrary but probably quite reasonable for most hash-table sizes
-            tile_set->m_AnimationIds.SetCapacity(37, n_animations);
-            for (uint32_t i = 0; i < n_animations; ++i)
+            uint32_t n_points = texture_set_ddf->m_CollisionHullPoints.m_Count / 2;
+            float recip_tile_width = 1.0f / (texture_set_ddf->m_TileWidth - 1);
+            float recip_tile_height = 1.0f / (texture_set_ddf->m_TileHeight - 1);
+            float* points = texture_set_ddf->m_CollisionHullPoints.m_Data;
+            float* norm_points = new float[n_points * 2];
+            for (uint32_t i = 0; i < n_points; ++i)
             {
-                dmhash_t h = dmHashString64(texture_set_ddf->m_Animations[i].m_Id);
-                tile_set->m_AnimationIds.Put(h, i);
+                norm_points[i*2] = (points[i*2]) * recip_tile_width - 0.5f;
+                norm_points[i*2+1] = (points[i*2+1]) * recip_tile_height - 0.5f;
             }
+            tile_set->m_HullSet = dmPhysics::NewHullSet2D(context, norm_points, n_points, hull_descs, n_hulls);
+            delete [] hull_descs;
+            delete [] norm_points;
         }
-        else
+
+        uint32_t n_animations = texture_set_ddf->m_Animations.m_Count;
+        tile_set->m_AnimationIds.Clear();
+        // NOTE: 37 is rather arbitrary but probably quite reasonable for most hash-table sizes
+        tile_set->m_AnimationIds.SetCapacity(37, n_animations);
+        for (uint32_t i = 0; i < n_animations; ++i)
         {
-            dmDDF::FreeMessage(texture_set_ddf);
+            dmhash_t h = dmHashString64(texture_set_ddf->m_Animations[i].m_Id);
+            tile_set->m_AnimationIds.Put(h, i);
         }
+
+        // This is a mapping from the single frame image names (animation ids) to the frame number
+        uint32_t n_image_name_hashes = texture_set_ddf->m_ImageNameHashes.m_Count;
+        tile_set->m_FrameIds.SetCapacity(dmMath::Max(1U, (n_image_name_hashes*2)/3), n_image_name_hashes);
+        for (uint32_t i = 0; i < n_image_name_hashes; ++i)
+        {
+            dmhash_t h = texture_set_ddf->m_ImageNameHashes[i];
+            tile_set->m_FrameIds.Put(h, i);
+        }
```

Comment:
This part is new. We keep a lookup table from animation name (E.g. "foo" from "path/foo.png") into the frame index.
This allows us to lookup images between atlases.